### PR TITLE
Remove legacy metadata endpoints

### PR DESCRIPTION
**Pull request by elcct**
### Description of change

This removes legacy `/metadata/*` endpoints. It should be merged on or after 17th October.

It also fixes tests for ordered metadata views and updates a comment.

### Checklist

* [x] Has a new newsfragment been created? Check [changelog/README.md](https://github.com/uktrade/data-hub-leeloo/blob/master/changelog/README.md) for instructions
* [x] Do any added or updated endpoints appear in the API documentation? See [docs/Maintaining the API documentation.md](https://github.com/uktrade/data-hub-leeloo/blob/develop/docs/Maintaining&#32;the&#32;API&#32;documentation.md) for more details
* [ ] Have any relevant search models been updated?
* [ ] Have any relevant fixtures (`fixtures/test_data.yaml`) been updated?
* [ ] Have any relevant select-/prefetch-related field lists in the views and search apps been updated?
* [ ] Has the admin site been updated (for new models, fields etc.)?
* [ ] Has the README been updated (if needed)?

### CODE REVIEW

File: datahub/metadata/test/test_views.py [hunk 1]
```diff
@@ -236,7 +236,7 @@ def test_team_view(metadata_client):
 
 
 class TestServiceView:
-    """Tests for the /metadata/service/ view."""
+    """Tests for the /v4/metadata/service/ view."""
```
Comment (reupen) [inline datahub/metadata/test/test_views.py]: Does the trailing slash need removing too?

```suggestion
    """Tests for the /v4/metadata/service view."""
```

File: datahub/metadata/views.py [hunk 2]
```diff
@@ -25,12 +24,14 @@ def _create_metadata_view(mapping, view_set_types=None, auth_attributes=None):
         'queryset': mapping.queryset,
         'serializer_class': mapping.serializer,
         '__doc__': f'List all {model._meta.verbose_name_plural}.',
+        'authentication_classes': (PaaSIPAuthentication, HawkAuthentication),
```
Comment (reupen) [inline datahub/metadata/views.py]: Should we put these new auth-related entries first (and keep `__doc__` last)?

**Comment by rafenden** [comment]
Could we please double check if no apps are using them? 

Looks like the market access are still pointing to the old endpoints:
https://github.com/uktrade/market-access-frontend/blob/develop/src/app/lib/metadata.js

**Comment by elcct** [comment]
Thank you for checking this out. The metadata you are referring to calls Market Access backend, not Data Hub. Market Access backend fetches Data Hub metadata and then serves it to front end. The change has been made so that Market Access backend uses new /v4 endpoints with Hawk, here: https://github.com/uktrade/market-access-api/commit/83e47c9a1b018650f43f7566b8fe8ad46f6d7ef6

**Comment by elcct** [comment]
@rafenden just confirmed that indeed Market Access is no longer using legacy metadata endpoints.

**Comment by rafenden** [comment]
> @rafenden just confirmed that indeed Market Access is no longer using legacy metadata endpoints.

Thank you for clarification 👍

**Pull request merged**